src/hex/grid/edge.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{Direction, Hex};
+use std::ops::Neg;
+
+use super::GridVertex;
+
+/// Hexagonal grid orientated edge representation
+#[derive(Debug, Clone, Copy)]
+pub struct GridEdge {
+    /// The coordinate of the edge
+    pub origin: Hex,
+    /// The direction the edge points towards
+    pub direction: Direction,
+}
+
+impl PartialEq for GridEdge {
```

Comment:
I fear that using partial eq for this is dangerous in combination with hashset / hasmaps
Also, depending on user code the grid edges might represent the 'internal part' of the edge
```
|      hex A     a|b      hex B     |
```

Reply:
Maybe a dedicated function would be safer ?
```
impl GridEdge {
    fn is_same_edge(&self, other: &Self) -> bool {
        self == other || self.flipped() == other
    }
}
```